limit_enforcement_plugins/src/ruckig_joint_limiter.cpp

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2021, PickNik Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// \authors Andy Zelenak
+
+#include "ruckig_joint_limiter/ruckig_joint_limiter.hpp"
+
+#include "limit_enforcement_plugins/joint_limits_rosparam.hpp"
+
+#include <string>
+
+namespace ruckig_joint_limiter
+{
+namespace
+{
+constexpr double DEFAULT_MAX_VELOCITY = 5;       // rad/s
+constexpr double DEFAULT_MAX_ACCELERATION = 10;  // rad/s^2
+constexpr double DEFAULT_MAX_JERK = 20;          // rad/s^3
+}  // namespace
+
+template <>
+bool RuckigJointLimiter<limit_enforcement_plugins::JointLimits>::init(
+  const std::vector<std::string> joint_names, const rclcpp::Node::SharedPtr & node,
```

Comment:
We should keep this implementation outside of here. This will result that each implemented plugin has to take care of reading the parameters. Ideally, users should only implement "enforce()" method. and maybe have short impolementation of "init()" to get some structures set-up.